### ISSUE TITLE
GraphQL Pro cursor breaks with AREL expression order

### ISSUE THREAD

**Issue by jscheid** [rmosolgo/graphql-ruby#3109]
**Describe the bug**

ActiveRecord accepts an Arel expression as an argument to `order`:

```ruby
[1] pry(main)> User.order(
[1] pry(main)*   Arel::Nodes::InfixOperation.new(  
[1] pry(main)*     '||',    
[1] pry(main)*     Arel::Nodes::InfixOperation.new(    
[1] pry(main)*       '||',      
[1] pry(main)*       Arel::Nodes::NamedFunction.new('btrim', [User.arel_table[:first_name]]),      
[1] pry(main)*     Arel::Nodes::SqlLiteral.new("' '")),        
[1] pry(main)*     Arel::Nodes::NamedFunction.new('btrim', [User.arel_table[:last_name]])    
[1] pry(main)*   ).desc    
[1] pry(main)* ).to_sql  
=> "SELECT \"users\".* FROM \"users\" ORDER BY btrim(\"users\".\"first_name\") || ' ' || btrim(\"users\".\"last_name\") DESC"
```

When an expression like this is used with GraphQL Pro cursor-based relation connection, an error is thrown:

```
     NoMethodError:
       undefined method `name' for #<Arel::Nodes::InfixOperation:0x0000561877b41b18>
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-pro-1.15.4/lib/graphql/pro/relation_connection/order.rb:47:in `block in normalize'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-pro-1.15.4/lib/graphql/pro/relation_connection/order.rb:33:in `each'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-pro-1.15.4/lib/graphql/pro/relation_connection/order.rb:33:in `normalize'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-pro-1.15.4/lib/graphql/pro/relation_connection.rb:16:in `new'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema/field/connection_extension.rb:54:in `after_resolve'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema/field.rb:735:in `block (2 levels) in with_extensions'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema/field.rb:732:in `each'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema/field.rb:732:in `each_with_index'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema/field.rb:732:in `block in with_extensions'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema.rb:112:in `after_lazy'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema/field.rb:731:in `with_extensions'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema/field.rb:666:in `public_send_field'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema/field.rb:555:in `block (2 levels) in resolve_field'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema.rb:112:in `after_lazy'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema/field.rb:545:in `block in resolve_field'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema.rb:112:in `after_lazy'
     # /home/vagrant/.rvm/gems/ruby-2.6.3/gems/graphql-1.11.2/lib/graphql/schema/field.rb:539:in `resolve_field'
```

Workaround is to cast the expression to a string first:

```ruby
[4] pry(main)> User.order(
[4] pry(main)*   Arel.sql(  
[4] pry(main)*     Arel::Nodes::InfixOperation.new(    
[4] pry(main)*       '||',      
[4] pry(main)*       Arel::Nodes::InfixOperation.new(      
[4] pry(main)*         '||',        
[4] pry(main)*         Arel::Nodes::NamedFunction.new('btrim', [User.arel_table[:first_name]]),        
[4] pry(main)*       Arel::Nodes::SqlLiteral.new("' '")),          
[4] pry(main)*       Arel::Nodes::NamedFunction.new('btrim', [User.arel_table[:last_name]])      
[4] pry(main)*     ).desc.to_sql      
[4] pry(main)*   )    
[4] pry(main)* ).to_sql  
=> "SELECT \"users\".* FROM \"users\" ORDER BY btrim(\"users\".\"first_name\") || ' ' || btrim(\"users\".\"last_name\") DESC"
```

**Versions**

`graphql` version: 1.11.2
`rails`: 6.0.3.2
`graphql-pro` version: 1.15.4

**Comment by rmosolgo** [comment]
Thanks for reporting this issue! 

Any reason not to upstream the workaround you mentioned, namely, calling `to_sql` on unrecognized Arel constructions?

**Comment by jscheid** [comment]
Good question, not 100% sure if there's a difference in how AR treats the two but I don't think there is.

**Comment by rmosolgo** [comment]
:+1: Just shipped 1.15.5 with this improvement, thanks for the report and the suggested solution!
